### PR TITLE
Fix deleting a worktree's preserved branch over SSH

### LINKED ISSUE #6418: [Bug] Dirty worktree deletion over ssh
<kbd>[Roderik](https://discord.com/channels/1492228674081263786/1520005076230475877/1520005076230475877)</kbd> [Roderik](https://discord.com/channels/1492228674081263786/1520005076230475877/1520005076230475877)  `via Discord`

Fails with this error after clicking the force delete button in the previous popup. The workspace does dissapear from orca but i think the worktree itself stays on the ssh host

### PR DESCRIPTION
**Pull request by brennanb2025**
## What changes

Force-deleting a worktree's **preserved branch over SSH** now succeeds. Previously it failed with `Error: git subcommand not allowed: worktree` and left the branch behind on the remote host.

**What does not change:** the local (non-SSH) preserved-branch delete path is untouched, and this does not widen the relay's read-only `git.exec` allowlist — it adds one narrow, purpose-built RPC instead.

## Root cause

When a dirty worktree with an unmerged branch is force-deleted, Orca removes the worktree but preserves the branch and offers a "force delete branch" action. Over SSH that action called `forceDeleteLocalBranch`, which runs `git worktree list`, `git update-ref -d`, and `git config --remove-section` through the relay's generic `git.exec` handler. That handler validates every call against a read-only allowlist, which rejects `worktree`/`update-ref`/`config` writes — so the first command (`git worktree list`) threw `git subcommand not allowed: worktree`. The normal worktree-removal path never hit this because it uses a dedicated relay RPC (`git.removeWorktree`) that runs git through the relay's internal executor.

## Design approach

Add a dedicated relay RPC `git.forceDeletePreservedBranch` that performs the delete sequence relay-side (checked-out guard → `update-ref -d` at the expected head → restore-on-race → best-effort `config --remove-section`), exactly mirroring how `git.removeWorktree` already bypasses the allowlist. The SSH runtime and IPC handlers now call `SshGitProvider.forceDeletePreservedBranch`, which issues that RPC. The relay reuses the existing branch-cleanup body shared with `removeWorktreeOp`; an optional error-mapper is applied to **only** the `update-ref -d` step so the actionable "branch changed after the workspace was deleted" message is produced for the force-delete path without remapping the checked-out-guard message or regressing `removeWorktreeOp`'s "branch preserved (not fully merged)" outcome.

Alternatives considered and rejected during design review: widening the `git.exec` allowlist (weakens the deliberately read-only SSH boundary — `update-ref` is a general ref-write primitive), and a generic privileged-op RPC (over-abstraction vs the codebase's one-RPC-per-op pattern).

## Implementation notes / deviations from design

- `forceDeletePreservedRelayBranch` validates the branch name (empty/NUL/leading-`-`) and expected head before any git call.
- Relay handler rejects empty/NUL `repoPath` and NUL `expectedHead` at the RPC boundary (defense-in-depth: an empty `repoPath` would resolve `cwd` to the relay's process directory).
- `SshGitProvider.forceDeletePreservedBranch` maps a JSON-RPC "method not found" (older relay binary) to an actionable reconnect message rather than silently failing; it does not fall back to `provider.exec` (that is the path that fails).
- Added `forceDeletePreservedBranch?` to the `IGitProvider` contract (optional, mirroring `renameCurrentBranch?`).
- No deviations from the reviewed design.

## Headline behavior status

**Implemented** in production code on both the runtime RPC (`OrcaRuntimeService.forceDeletePreservedBranch`) and the IPC (`worktrees:forceDeletePreservedBranch`) SSH branches. Not fixture-only or dormant.

## Validation

**Design review:** ran to clean (no P0/P1; direction confirmed).

**Completeness verification:** confirmed every requirement and edge case implemented; flagged a missing runtime/IPC unit-test for the SSH path, which was then added.

**Perf audit:** no actionable findings. Net improvement — over SSH the old path issued one `git.exec` round-trip per git command; the new path issues a single RPC that runs all commands relay-side. No new polling, render, subscriber, or startup work. Rare one-shot user action; deterministic command/RPC-count tests are sufficient (added), no runtime measurement warranted.

**Code-review loop:** two independent reviewers per round. Round 1 surfaced two actionable items (validate empty/NUL `repoPath`+`expectedHead` at the RPC boundary; add relay dispatch-contract coverage for the new RPC) plus a contract nit (`IGitProvider`); a flagged test-file-length item was a false positive (test files allow 800 lines). All fixed. Round 2: both reviewers returned clean.

**brennan-test-changes (SSH end-to-end): verdict LAND.** Exercised the real flow against a throwaway Dockerized Linux SSH target (debian-slim + openssh + git, SSH on 127.0.0.1:2222; repo seeded via local git-bundle fallback since the container had no GitHub auth; Electron launched from this worktree with an isolated user-data dir and CDP port, identity verified before acting). Created a worktree over SSH on branch `test-6418-run2-...`, made it dirty with an unmerged commit, force-deleted the worktree (branch preserved at HEAD `45cc64e2…`), then invoked `worktrees:forceDeletePreservedBranch` (the exact IPC the force-delete button calls), which returned `{ deleted: true }`. Remote proof on `/work/demo-project`:

- before: `git show-ref` had `refs/heads/test-6418-run2-...` at `45cc64e2…`
- after: `git branch --list 'test-6418-*'` → **empty**; `git show-ref | grep test-6418` → **empty**; `git worktree list` → only `/work/demo-project [main]`
- **zero** `git subcommand not allowed` matches in the Playwright console, Electron main log, or remote relay log; the relay log confirms the preserved-branch path ran and completed without falling back through `git.exec`.

Relay-level real-git tests additionally prove the same deterministically (ref deleted at the expected head; refused when the head moved).

**Static checks / focused tests:** `pnpm run typecheck` passes; `oxlint` clean on all touched files; 809 tests pass across `git-handler.test.ts`, `git-handler-branch-cleanup.test.ts`, `ssh-git-provider.test.ts`, `orca-runtime.test.ts`, `worktrees.test.ts`. New tests cover: relay delete-at-expected-head success / ref-moved rejection / checked-out guard / concurrent-checkout restore / config-failure swallow / invalid input; that `removeWorktreeOp` still propagates the raw error (unchanged); provider RPC params + old-relay mapping + rethrow; runtime and IPC SSH branches route to the new provider method; and dispatcher-level real-git happy-path/ref-moved/empty-repoPath cases.

## Out of scope

The post-delete fork-remote cleanup (`cleanupUnusedWorktreePushTargetRemoteSsh`) also routes a `git remote remove` through the allowlist, but it is best-effort (try/catch that only warns), affects only fork-PR worktrees, is identical in the normal removal path, and is not part of the reported dirty-delete failure. Noted for a possible follow-up.

## Remaining risks / accepted gaps

During SSH validation the renderer's force-delete-preserved-branch **button** did not surface for this particular force-remove flow (the toast/panel hosting it is gated on a code path that doesn't fire for `force=true` removal of a dirty unmerged worktree on this baseline). Validation drove the exact IPC that button calls (`worktrees:forceDeletePreservedBranch`) with the same args, exercising the entire fixed relay/RPC path; surfacing the button is a separate concern, out of scope for this PR. No other accepted gaps.

Fixes #6418

Made with [Orca](https://github.com/stablyai/orca) 🐋

### CODE REVIEW

File: src/relay/git-handler-branch-cleanup.ts [hunk 1]
```diff
@@ -26,16 +26,42 @@ export async function deleteAlreadyMergedRelayBranchAfterSafeDeleteFailure(
...
+    throw new Error('Invalid branch name for preserved branch delete.')
+  }
+  if (!expectedHead) {
+    throw new Error('Expected branch head is required for preserved branch delete.')
+  }
+  await deleteRelayBranchAtExpectedHead(git, repoPath, branchName, expectedHead, () => {
+    return new Error(
+      `Local branch "${branchName}" changed after the workspace was deleted. Review it before deleting it.`
+    )
+  })
```
[the remark this answers is not in the document]
Comment (brennanb2025) [reply src/relay/git-handler-branch-cleanup.ts:45]: This is a deliberate parity choice, not a regression. The relay sequence mirrors the local `forceDeleteLocalBranch` (src/main/git/worktree.ts), which wraps the same `update-ref -d` step in an identical catch-all and throws the same "changed after the workspace was deleted" message. Making the SSH path gate on a parsed compare-and-swap mismatch would diverge it from the local behavior and depend on locale/version-specific `update-ref` stderr (your own analysis shows the wording varies), which is brittle.

In practice the ref was just preserved by the immediately-preceding removal, so it exists; the realistic failure mode this guards is a moved head, which is exactly the compare-and-swap mismatch. The mapper is also already scoped to ONLY the first `update-ref -d` call — the checked-out guard and the restore `update-ref` keep their own distinct/raw errors, and `removeWorktreeOp` still passes no mapper so its raw-error path is unchanged. Keeping local/SSH parity here is intentional.
